### PR TITLE
[mlir][tensor] Add a pattern to simplify tensor.unpack to collpase shape

### PR DESCRIPTION
**Pull request by hanhanW**

### CODE REVIEW

**Comment by hanhanW** [comment]
It depends on https://github.com/llvm/llvm-project/pull/76606, please take a look at https://github.com/llvm/llvm-project/pull/76607/commits/8190cd1b2846a906be53eb281135c047afe4b7c8 or wait until the PR is landed.

**Review by chelini — APPROVED** [review]
Thanks!

**Pull request merged**
